zircon.core/src/commonMain/kotlin/org/hexworks/zircon/internal/fragment/impl/table/DefaultTable.kt

```diff
@@ -0,0 +1,127 @@
+package org.hexworks.zircon.internal.fragment.impl.table
+
+import org.hexworks.cobalt.databinding.api.extension.toProperty
+import org.hexworks.cobalt.databinding.api.property.Property
+import org.hexworks.cobalt.databinding.api.value.ObservableValue
+import org.hexworks.zircon.api.Components
+import org.hexworks.zircon.api.component.AttachedComponent
+import org.hexworks.zircon.api.component.Component
+import org.hexworks.zircon.api.component.HBox
+import org.hexworks.zircon.api.component.VBox
+import org.hexworks.zircon.api.data.Size
+import org.hexworks.zircon.api.fragment.Table
+import org.hexworks.zircon.api.uievent.MouseEventType
+import org.hexworks.zircon.api.uievent.UIEventPhase
+import org.hexworks.zircon.api.uievent.UIEventResponse
+
+/**
+ * The **internal** default implementation of [Table].
+ */
+class DefaultTable<M: Any>(
+    private val data: List<M>,
+    private val columns: List<TableColumn<M, *, *>>,
+    /**
+     * The height this fragment will use. Keep in mind that the first row will be used as header row.
+     */
+    fragmentHeight: Int,
+    private val rowSpacing: Int = 0,
+    private val colSpacing: Int = 0
+): Table<M> {
+
+    init {
+        require(data.isNotEmpty()) {
+            "A table may not be empty! Please feed it some data to display."
+        }
+        require(columns.isNotEmpty()) {
+            "A table must have at least one column."
+        }
+        val minHeight = 2
+        require(fragmentHeight >= minHeight) {
+            "A table requires a height of at least $minHeight."
+        }
+    }
+
+    private val selectedElement: Property<M> = data.first().toProperty()
+
+    override val selectedRowValue: ObservableValue<M> = selectedElement
+
+    override val selectedRow: M
+        get() = selectedRowValue.value
+
+    override val size: Size = Size.create(
+        width = columns.sumBy { it.width } + ((columns.size - 1) * colSpacing),
+        height = fragmentHeight
+    )
+
+    override val root: VBox = Components
+        .vbox()
+        .withSpacing(0)
+        .withSize(size)
+        .build()
+
+    private val currentRows: MutableList<AttachedComponent> = mutableListOf()
+
+    init {
+        val headerRow = headerRow()
+        root
+            .addComponents(
+                headerRow,
+                dataPanel(Size.create(size.width, size.height - headerRow.height))
```

Comment:
Minor: isn't there a better way to create this Size? `withHeight` or `withRelativeHeight` or something?